src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/change/ScheduledStateTransitionHandler.java

```diff
@@ -25,17 +32,26 @@
     @Override
     public void handleEventLifecycle(CollectionExerciseEventPublisher.MessageType change, Event event) throws CTPException {
         CollectionExercise collectionExercise = event.getCollectionExercise();
+        CollectionExerciseDTO.CollectionExerciseEvent ceEvent = null;
 
-        if (this.eventService.isScheduled(collectionExercise.getId())){
-            this.collectionExerciseService.transitionCollectionExercise(collectionExercise,
-                    CollectionExerciseDTO.CollectionExerciseEvent.EVENTS_ADDED);
-            // This is a bit of a kludge on account of SCHEDULED not being a proper state.  If the CI & sample are
-            // already loaded, loading the events may trigger a transition straight through to READY_FOR_REVIEW
-            // Performing this call here will force that
-            this.collectionExerciseService.transitionScheduleCollectionExerciseToReadyToReview(collectionExercise);
-        } else {
-            this.collectionExerciseService.transitionCollectionExercise(collectionExercise,
-                    CollectionExerciseDTO.CollectionExerciseEvent.EVENTS_DELETED);
+        try {
+            if (this.eventService.isScheduled(collectionExercise.getId())) {
+                ceEvent = CollectionExerciseDTO.CollectionExerciseEvent.EVENTS_ADDED;
+                this.collectionExerciseService.transitionCollectionExercise(collectionExercise, ceEvent);
+                // This is a bit of a kludge on account of SCHEDULED not being a proper state.  If the CI & sample are
+                // already loaded, loading the events may trigger a transition straight through to READY_FOR_REVIEW
+                // Performing this call here will force that
+                this.collectionExerciseService.transitionScheduleCollectionExerciseToReadyToReview(collectionExercise);
+            } else {
+                ceEvent = CollectionExerciseDTO.CollectionExerciseEvent.EVENTS_DELETED;
+                this.collectionExerciseService.transitionCollectionExercise(collectionExercise, ceEvent);
+            }
+        } catch(CTPException e) {
+            // As the events are deliberately fired indiscriminately (i.e. the collection exercise state is not
+            // checked first) there is a reasonable likelihood that the transition will fail harmlessly. Hence this
+            // exception is being logged as a warning minus the stack trace
+            log.warn("Collection exercise {} failed to handle state transition {} - {} ({})",
```

Comment:
What's the reason for surprising the stacktrace?

Reply:
Reduce noise in the logs - these can happen alot and they are mostly harmless